packages/backend/src/apps/m365-excel/common/create-plumber-folder.ts

```diff
@@ -13,21 +37,38 @@ export async function createPlumberFolder(
   const tenant = getM365TenantInfo(tenantKey)
 
   // Create folder
-  const createFolderResult = await $.http.post<{ id: string }>(
-    '/v1.0/sites/:sharePointSiteId/drive/root/children',
-    {
-      name: $.user.email,
-      folder: {},
-    },
-    {
-      urlPathParams: {
-        sharePointSiteId: tenant.sharePointSiteId,
+  let folderId = ''
+  try {
+    const createFolderResult = await $.http.post<{ id: string }>(
+      '/v1.0/sites/:sharePointSiteId/drive/root/children',
+      {
+        name: $.user.email,
+        folder: {},
       },
-    },
-  )
-  const folderId = createFolderResult.data.id
+      {
+        urlPathParams: {
+          sharePointSiteId: tenant.sharePointSiteId,
+        },
+      },
+    )
+    folderId = createFolderResult.data.id
+  } catch (error) {
+    const folderAlreadyExists =
+      error instanceof HttpError &&
+      tryParseGraphApiError(error)?.code === 'nameAlreadyExists'
+    if (!folderAlreadyExists) {
+      throw error
+    }
+
+    folderId = await checkForExistingFolder($, tenant)
+
+    if (!folderId) {
+      throw error
+    }
```

Comment:
can we shift this outside of the catch clause, so it handles for empty folderId in the event where the http.post actually succeeds for but for some reason folderId is empty

Reply:
good call, done